### ISSUE TITLE
eliminate redundant links in the people listing: 2

### ISSUE THREAD

**Issue by paregorios** [isawnyu/isaw.web#335]
At present, each entry in a "people listing" contains two links to the corresponding profile (one wrapped around the image and the other wrapped around the individual's name):

```html
<div class="facultyListing">
  <a href="http://isaw.nyu.edu/people/faculty/isaw-faculty/lorenzo-dalfonso">
    <span class="facultyImage">
      <img src="http://isaw.nyu.edu/people/faculty/isaw-faculty/lorenzo-dalfonso/@@images/Image/mini">
    </span>
  </a>
  <div class="facultyInfo">
    <h2>
      <a href="http://isaw.nyu.edu/people/faculty/isaw-faculty/lorenzo-dalfonso">Lorenzo d'Alfonso</a>
    </h2>
    <p class="facultyPosition"></p><p>Associate Professor of Western Asian Archaeology and History</p>
    <p class="facultyEmail">lda5@nyu.edu</p>
  </div>
</div>
```

These adjacent, redundant links are an accessibility problem, so we need to reconfigure the markup in order to get a single link encompassing both the image and name. Something like this:

```html
<div class="facultyListing">
  <div class="facultyInfo">
    <a href="http://isaw.nyu.edu/people/faculty/isaw-faculty/lorenzo-dalfonso">
      <span class="facultyImage">
        <img src="http://isaw.nyu.edu/people/faculty/isaw-faculty/lorenzo-dalfonso/@@images/Image/mini">
      </span>
      <h2>Lorenzo d'Alfonso</h2>
    </a>
    <p class="facultyPosition"></p><p>Associate Professor of Western Asian Archaeology and History</p>
    <p class="facultyEmail">lda5@nyu.edu</p>
  </div>
</div>
```

## Caveats

 - It is important that we don't change visual look and feel, so it's possible that some CSS tweaks will be needed in order to address visual formatting of the revised structure
 - Simplifying the HTML would be fine too, as long as we get accessible and stay that way (and preserve visual look and feel). So, e.g., dropping the ```<div><div>``` encapsulation could be considered.

**Comment by paregorios** [comment]
Staging demonstrates the main aspect of the desired outcome: a single link, rather than multiple redundant ones, for each person; however, there are a couple of rough edges.

## Look and feel has changed

Compare the "ISAW Faculty" page on production (first) to staging (second): the position is getting bolded on staging where it should be normal weight as on production.

<img alt="screen shot 2018-09-26 at 5 15 33 am" src="https://user-images.githubusercontent.com/263285/46073799-86054380-c14b-11e8-8d59-b4507ae9b7db.png">

<img alt="screen shot 2018-09-26 at 5 19 01 am" src="https://user-images.githubusercontent.com/263285/46073875-ba78ff80-c14b-11e8-8999-cc07d50592d1.png">

The bolding on staging is occurring because there is a CSS rule in ```isaw.theme/isaw/theme/static/theme.css``` that does the deed and the position text is finally appearing inside ```<p class="facultyPosition">``` instead of after it:

```css
.section-people #portal-columns .facultyListing .facultyInfo .facultyPosition,
.section-people #content .facultyListing .facultyInfo . {font-size:14px; font-weight:bold; }
```

But because we've become accustomed to seeing the listing as it appears presently on production, we should **eliminate this CSS rule from the theme entirely**. 

## Unsafe content in Profile:Faculty Titles field causes problems

It's not evident in the "ISAW Faculty" page on staging, but if one looks at the "Staff" page one can see that something odd is happening to some of the entries:

<img width="725" alt="screen shot 2018-09-26 at 5 32 42 am" src="https://user-images.githubusercontent.com/263285/46074584-a2a27b00-c14d-11e8-84c5-84d178560e52.png">

The "Faculty Titles" field on the ```Profile``` content item is a rich text field, and therefore can contain links. When there's a link in that field (e.g., Diane Bennett and Andrea Chang in the above screenshot), it's coming through in the "People Listing" and forcing the DOM to cope by breaking apart the single, overarching ```<a>``` element into multiples (undermining the main focus of this ticket). The resulting HTML structure then opens things up for CSS rules to do fun things with look and feel too.

I think the best solution is for the "Page Listing" to **sanitize the contents of ```Profile:Faculty Titles```** down to plain text (not ASCII!, but no markup) before populating ```<div class="facultyPosition">```.

**Comment by paregorios** [comment]
Woops. It looks like *some* of the markup in ```Profile:Faculty Titles``` is potentially valuable for People listing. See, e.g.,:

<img width="698" alt="screen shot 2018-09-26 at 5 53 41 am" src="https://user-images.githubusercontent.com/263285/46075591-9370fc80-c150-11e8-9df2-a7a5ebb6be93.png">

That ```<br>``` between "Leon Levy Director" and the rest of the position title is actually important even in the people listing.

So, now the question is, how do we avoid the breakage if somebody's put a link into that field, but still keep other formatting?

**Comment by jessesnyder** [comment]
Notes:
- **bolding of the faculty position**: the original intent in the markup and CSS was for this to be bold. It was/is failing because of the problem with nested `<p>` tags getting automatically un-nested by the browser (Alec's guess here was correct). Easy enough to remove the bolding if maintaining the status quo is the priority.
- **Embedded markup in `<div class="facultyPosition" />`**: I think I can just go back to my first version of the markup, which was slightly more conservative in regards to the scope of the `<a />`. It can contain just the image and the title, and all we lose is the entire entry acting as a link and having a hover state.

**Comment by jessesnyder** [comment]
@paregorios Back over to you. <whack!>

**Comment by paregorios** [comment]
That's got it, thanks! Please merge to master and deploy to production.
